public/static/app/perf/perf.js

```diff
@@ -659,24 +644,30 @@ function TrendSamples(samples){
     }
   }
 
-  for(key in this.seriesByName){
+  for(let key in this.seriesByName){
     this.seriesByName[key] = _.sortBy(this.seriesByName[key], 'order');
     this.testNames.unshift(key);
   }
 
-  for(var i=0; i < this.testNames.length; i++){
+  for(let i = 0; i < this.testNames.length; i++) {
     //make an index for commit hash -> sample for each test series
     var k = this.testNames[i];
+    // FIXME Unknown behavior (coma operator after _groupBy stmt)
     this._sampleByCommitIndexes[k] = _.groupBy(this.seriesByName[k], "revision"), function(x){return x[0]};
-    for(t in this._sampleByCommitIndexes[k]){
+    for(let t in this._sampleByCommitIndexes[k]){
       this._sampleByCommitIndexes[k][t] = this._sampleByCommitIndexes[k][t][0];
     }
   }
 
   // Returns a list of samples for a given test, sorted in the order that they were committed.
   this.tasksByCommitOrder = function(){
-    if(!this._tasks){
-      this._tasks = _.sortBy(_.uniq(_.flatten(_.values(this.seriesByName)), false,  function(x){return x.task_id}), "order");
+    if (!this._tasks) {
+      this._tasks = _.chain(this.seriesByName)
+        .values()
+        .flatten()
+        .uniq(false, d => d.task_id)
+        .sortBy('order')
+        .value()
```

Comment:
Is the final .value() needed? It wasn't there before

Reply:
It does. I changed syntax to `_.chain`